### PR TITLE
Ts/add to mira model edit

### PR DESCRIPTION
**Pull request by Tom-Szendrey**
Fixing up the context i created earlier by adding missing functions as well as adding the `loop` into the tools to allow them to correctly output code_cells

Renaming the intercepts to all include `_request` 

https://github.com/DARPA-ASKEM/beaker-kernel/assets/17088680/e52bc9e1-9cbe-408f-91d2-b8d4f69fe54c

### CODE REVIEW

File: src/askem_beaker/contexts/mira_model_edit/agent.py [hunk 1]
```diff
@@ -87,84 +89,12 @@ async def add_template(self, name: str, subject: str, outcome: str, expr: str, m
             expr (str): the mathematical rate law for the transition.
             name (str): the name of the transition
         """
-        code = agent.context.get_code("add_template", {"model": model, "subject": subject, "outcome": outcome, "expr": expr, name: "name"})
-        return json.dumps(
+        code = agent.context.get_code("add_template", {"model": model, "subject": subject, "outcome": outcome, "expr": expr, "name": name})
+        loop.set_state(loop.STOP_SUCCESS)
+        return json.dumps( # <--- Is this returning to archytas? Why does this action not mean it sends a response with code_cell
```
Comment (mwdchang) [inline src/askem_beaker/contexts/mira_model_edit/agent.py:94]: Are we doing something about the comment here?

File: src/askem_beaker/contexts/mira_model_edit/context.py [hunk 2]
```diff
@@ -34,11 +34,9 @@ def __init__(self, beaker_kernel: "LLMKernel", subkernel: "BaseSubkernel", confi
         self.reset()
         self.auth = get_auth()
         super().__init__(beaker_kernel, subkernel, self.agent_cls, config)
-
-    def reset(self):
-        pass
-        
+    
     async def setup(self, config, parent_header):
+        logger.error(f"performing setup...")
```
Comment (mwdchang) [inline src/askem_beaker/contexts/mira_model_edit/context.py:39]: Flagging this as an error might be a little excessive

File: src/askem_beaker/contexts/mira_model_edit/context.py [hunk 3]
```diff
@@ -117,45 +118,40 @@ async def reset_request(self, message):
...
+    async def replace_template_name_request(self, message):
         content = message.content
 
         model = content.get("model")
         old_name  = content.get("old_name")
         new_name = content.get("new_name")
 
+        # logger.error("Replace template name request")
+
+        codeObj = self.agent.replace_template_name(model,old_name,new_name)
```
Comment (mwdchang) [inline src/askem_beaker/contexts/mira_model_edit/context.py:130]: Nitpick: You have a couple of spots where you do camels, may want to do snakes to be more consistent with the rest of the repo.

File: src/askem_beaker/contexts/mira_model_edit/agent.py (hunk 1 above)
Comment (Tom-Szendrey) [reply src/askem_beaker/contexts/mira_model_edit/agent.py:94]: Thanks for catching

File: src/askem_beaker/contexts/mira_model_edit/context.py (hunk 2 above)
Comment (Tom-Szendrey) [reply src/askem_beaker/contexts/mira_model_edit/context.py:39]: LOL Fair I just grabbed the same status in the newest context I could find model_config_edit 
ill update this

**Pull request merged**